frontend/packages/topology/src/components/workload/useResourceWatcher.ts

```diff
@@ -0,0 +1,17 @@
+import * as React from 'react';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { K8sResourceCommon } from '@console/internal/module/k8s';
+import { getResourceFromWatchResults } from './utils';
+
+export const useResourceWatcher = <R extends K8sResourceCommon>(resourceData: R) => {
+  const watchKind = resourceData.kind;
+  const { namespace, uid } = resourceData.metadata;
+  const watchedResourceObj = React.useMemo(() => ({ isList: true, kind: watchKind, namespace }), [
+    watchKind,
+    namespace,
+  ]);
```

Comment:
Instead of watching all the resources (Deployments) and having a filtering logic based on uid, you could only watch for that specific resource by adding the `name` to the watchObject and setting `isList` to false
```suggestion
  const watchedResourceObj = React.useMemo(() => ({ isList: false, kind: watchKind, namespace, name }), [
    watchKind,
    namespace,
  ]);
```